### ISSUE TITLE
Puts and Posts failing when using "ember server --proxy"

### ISSUE THREAD

**Issue by cah-briangantzler** [sglanzer/ember-cli-blanket#46]
I upgraded to 0.5.0 and now all the puts and posts are cancelled when going through a proxy similar to this [post](http://stackoverflow.com/questions/23856465/ember-cli-proxy-works-for-get-but-fails-on-put-and-post) from some time ago. Looking at the changes from version 4 to version 5, there is a middleware function in 5 index.js that includes the line 
```
app.use(bodyparser.urlencoded({ extended: true });
```
 If I comment out this line, my proxy starts to work again, however the lcovOptions outfile appears to no longer be produced, which is the reason we upgraded to 5.  Could you look into getting ember-blanket-cli to play nice with the proxy please.

**Comment by jschilli** [comment]
@cah-briangantzler I've removed that line locally, restarted my app (with a `--proxy http://localhost:9292`) and the following blanket.options

```
/*globals blanket */

var options = {
   modulePrefix: "todomvc-ember-cli",
   filter: "//.*todomvc-ember-cli/.*/",
   antifilter: "//.*(tests|template).*/",
   loaderExclusions: [],
   enableCoverage: true,
   cliOptions: {
    outputFile: 'lcov.dat',
    reporters: ['lcov']
  }

};
if (typeof exports === 'undefined') {
  blanket.options(options);
} else {
  module.exports = options;
}
```

and I do get the `lcov.dat` file outputted correctly.

IIRC the `bodyparser.urlencoded` was in an early rev of this feature when I wasn't `POST`ing the data as JSON.

Can you verify that the POST is failing in a browser?  Any other diagnostics?

**Comment by jschilli** [comment]
which is to say that I'm happy to push a change to remove `bodyparser.urlencoded` after some testing, but we should figure out if that's going to work for you.

**Comment by cah-briangantzler** [comment]
There may be an issue with our blanket options. We have below. Im not sure why they chose to put the outfile in a lcovOptions hash. I will try moving it to the same level you have and try it

```
/* globals module, blanket */

var options = {
   modulePrefix: "vitalpath-ember-cli",
   filter: "//.*vitalpath-ember-cli/.*/",
   antifilter: ["//.*(tests).*/", "//.*vitalpath-ember-cli/templates/","//.*vitalpath-ember-cli/config/"],

   loaderExclusions: [],

   enableCoverage: true,
   cliOptions: {
        lcovOptions: {
            outputFile: 'log/lcov.dat'
        },
        reporters: ['lcov']
   }
};
if (typeof exports === 'undefined') {
    blanket.options(options);
} else {
    module.exports = options;
}
```

Thanks for the speedy responce

**Comment by jschilli** [comment]
`outputfile` is nested because we’re paving the way for multiple reporters and options per reporter

less future change with that (although admittedly awkward without current functionality for such)

There should be a default output file for each reporter - I can look @ that seperately.

Also, see #44 (https://github.com/sglanzer/ember-cli-blanket/issues/44 <https://github.com/sglanzer/ember-cli-blanket/issues/44>) - if you’re running headlessly you need to update your testem.json (or a variant thereof) to get coverage data written.

Easiest way to confirm is to run in the browser.

-jeff

**Comment by cah-briangantzler** [comment]
We have retest again and I must have not had something saved. Commenting out that line and having the blanket options as I have above (with nesting) does create the files as needed and fixes my problem.

**Comment by jschilli** [comment]
Thanks for verifying.

I will push a change to remove the urlencoded parser.

**Comment by cah-briangantzler** [comment]
Got busy at work, sorry. Finally got around to trying this. Our original problem was, when we authenticate to j_security_check as a POST through the ember server proxy, it was getting cancelled. Removing the line that we talked about started that working, however once in the app and I try to POST to an actual rest service, POSTs and PUTs are still getting cancelled. I thought I had tested this prior to posting the initial issue and suggested fix. Turns out, with 5.0 and the line removed, it gets past the POST to j_security_check but no actual rest POST. :( So of course, with 5.1 same behaviour. I find that removing the bodyparser.json() does indeed get me past the POST problem, but blanket no longer works. 

I have no suggestion on a fix at this point, but as far as I can tell, ember-blanket-cli no longer seems to work with the ember server proxy.
